lib/driver.js

```diff
@@ -1,407 +1,388 @@
+      any: consistencies.any
+    };
 
-DatastaxDriver.prototype.init = function init(context) {
-  if (!context) {
-    throw new Error('missing context object');
-  }
-  if (!context.config) {
-    throw new Error('missing context.config object');
+    this._init(context || {});
   }
 
-  /* istanbul ignore next: no benefit of testing coalesce */
-  this.config = context.config || {};
-  this.logger = context.logger || fakeLogger;
-  this.metrics = context.metrics;
-  if (context.config.queryDirectory) {
-    this.queryCache = new QueryCache({ queryDirectory: context.config.queryDirectory });
+  _init(context) {
+    if (!context.config) {
+      throw new Error('missing context.config object');
+    }
+
+    this.logger = context.logger || fakeLogger;
+    this.metrics = context.metrics;
+    this._initDriverConfig(context.config);
+
+    if (this.config.queryDirectory) {
+      this.queryCache = new QueryCache({ queryDirectory: this.config.queryDirectory });
+    }
+
+    this._initProviderOptions();
+    this.keyspace = this.config.keyspace;
+    this._initPoolConfig();
+    this._initConnectionResolver(context);
+
+    this.pools = {};
+    this.resultTransformers = context.resultTransformers || [];
   }
 
-  this.initProviderOptions(this.config);
-  this.keyspace = context.config.keyspace;
-  this.initPoolConfig();
-  this.initConnectionResolver(context);
+  _initDriverConfig(config) {
+    config.retryOptions = config.retryOptions || { retries: 0 };
+    const retryDelay = Math.max(config.retryDelay || 100, 0);
+    if (config.numRetries) {
+      config.retryOptions = {
+        retries: Math.max(config.numRetries, 0),
+        factor: 1,
+        minTimeout: retryDelay,
+        maxTimeout: retryDelay,
+        randomize: false
+      };
+    }
+    config.supportsPreparedStatements = true;
+    this.config = config;
+  }
 
-  this.pools = {};
-  this.resultTransformers = context.resultTransformers || [];
-};
+  _initProviderOptions() {
+    setHelenusOptions(this.config);
+  }
 
-DatastaxDriver.prototype.initProviderOptions = function init(config) {
-  setHelenusOptions(config);
-  config.supportsPreparedStatements = true;
-};
+  _initPoolConfig() {
+    this.poolConfig = {
+      consistencyLevel: this.consistencyLevel.one
+    };
+    this.poolConfig = _.extend(this.poolConfig, this.config);
 
-DatastaxDriver.prototype.initPoolConfig = function () {
-  this.poolConfig = {
-    consistencyLevel: this.consistencyLevel.one
-  };
-  this.poolConfig = _.extend(this.poolConfig, this.config);
+    if (typeof this.config.consistency === 'string') {
+      var level = this.config.consistency;
 
-  if (typeof this.config.consistency === 'string') {
-    var level = this.config.consistency;
+      if (typeof this.consistencyLevel[level] !== 'undefined') {
+        this.poolConfig.consistencyLevel = this.consistencyLevel[level];
+      } else {
+        throw new Error('Error: "' + level + '" is not a valid consistency level');
```

Comment:
Since you're already modernizing code, can use template strings here

Reply:
@scommisso: done